packages/devtools_app/lib/src/flutter/table.dart

```diff
@@ -0,0 +1,395 @@
+    // Size the table to only fit the items that are visible.
+    final flattenedList = _flattenExpandedTree(filter: (n) => n.isExpanded);
+    final root = flattenedList[0];
+    TreeNode deepest = root;
+    for (var node in flattenedList) {
+      if (node.level > deepest.level) {
+        deepest = node;
+      }
+    }
+    final widths = <double>[];
+    for (ColumnData<T> column in widget.columns) {
+      double width = column.getNodeIndentPx(deepest).toDouble();
+      if (column.fixedWidthPx != null) {
+        width += column.fixedWidthPx;
+      } else {
+        // TODO(djshuckerow): measure the text of the longest content
+        // to get an idea for how wide this column should be.
+        // Text measurement is a somewhat expensive algorithm, so we don't
+        // want to do it for every row in the table, only the row
+        // we are confident is the widest.  A reasonable heuristic is the row
+        // with the longest text, but because of variable-width fonts, this is
+        // not always true.
+        width += defaultColumnWidth;
+      }
+      widths.add(width);
+    }
+    return widths;
+  }
+
+  /// Resizes the table when the tree structure has been updated.
+  ///
+  /// [TreeNodeWidgets] call this method when they have told a [TreeNode]
+  /// to expand or collapse.
+  void refreshTree() {
+    setState(() {
+      columnWidths = _computeTableWidth();
+      _widthTween = Tween<double>(
+        begin: _widthTween.evaluate(_resizeController),
+        end: tableWidth,
+      );
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SingleChildScrollView(
+      scrollDirection: Axis.horizontal,
+      child: AnimatedBuilder(
+        animation: _resizeController,
+        builder: (context, child) {
+          return SizedBox(
+            width: max(
+              MediaQuery.of(context).size.width,
+              _widthTween.evaluate(_resizeAnimation),
+            ),
+            child: child,
+          );
+        },
+        child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
+          _TreeNodeWidget(
+            node: null,
+            tableState: this,
+          ),
+          Expanded(
+            child: ListView.custom(
+              childrenDelegate: SliverChildListDelegate([
+                for (var node in _flattenedList)
+                  _TreeNodeWidget(
+                    tableState: this,
+                    node: node,
+                  ),
+              ]),
+            ),
+          ),
+        ]),
+      ),
+    );
+  }
+}
+
+class _TreeNodeWidget<T extends TreeNode<T>> extends StatefulWidget {
+  const _TreeNodeWidget({
+    Key key,
+    @required this.node,
+    @required this.tableState,
+  }) : super(key: key);
+
+  final _TreeTableState<T> tableState;
+  final T node;
+
+  @override
+  _TreeNodeState createState() => _TreeNodeState<T>();
+}
+
+class _TreeNodeState<T extends TreeNode<T>> extends State<_TreeNodeWidget<T>>
```

Comment:
this class has some interesting functionality we should start reusing in the inspector tree to make it more nicely animated.

Reply:
I'm thinking that we'll want a common widget defining how we draw and animate all rows in our nested tables.